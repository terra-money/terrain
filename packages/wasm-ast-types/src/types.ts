export declare type fieldType = 'Long' | 'Coin' | 'Duration' | 'Height' | string;

export interface Field {
    name: string;
    type: fieldType;
    node: any;
};
export interface Interface {
    name: string;
    fields: Field[]
};

export interface QueryMsg {
    $schema: string;
    title: "QueryMsg";
    oneOf?: any;
    allOf?: any;
    anyOf?: any;
}

export interface ExecuteMsg {
    $schema: string;
    title: "ExecuteMsg" | "ExecuteMsg_for_Empty";
    oneOf?: any;
    allOf?: any;
    anyOf?: any;
}
