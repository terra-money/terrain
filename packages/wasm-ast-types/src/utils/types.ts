import * as t from '@babel/types';
import { camel } from 'case';
import { propertySignature } from './babel';

const getTypeFromRef = ($ref) => {
    switch ($ref) {
        case '#/definitions/Binary':
            return t.tsTypeReference(t.identifier('Binary'))
        default:
            if ($ref?.startsWith('#/definitions/')) {
                return t.tsTypeReference(t.identifier($ref.replace('#/definitions/', '')))
            }
            throw new Error('what is $ref: ' + $ref);
    }
}

const getArrayTypeFromRef = ($ref) => {
    return t.tsArrayType(
        getTypeFromRef($ref)
    );
}
const getArrayTypeFromType = (type) => {
    return t.tsArrayType(
        getType(type)
    );
}


export const getType = (type) => {
    switch (type) {
        case 'string':
            return t.tsStringKeyword();
        case 'boolean':
            return t.tSBooleanKeyword();
        case 'integer':
            return t.tsNumberKeyword();
        // case 'object':
        // return t.tsObjectKeyword();
        default:
            throw new Error('contact maintainers [unknown type]: ' + type);
    }
}


export const getPropertyType = (schema, prop) => {
    const props = schema.properties ?? {};
    let info = props[prop];

    let type = null;
    let optional = !schema.required?.includes(prop);

    if (info.allOf && info.allOf.length === 1) {
        info = info.allOf[0];
    }

    if (typeof info.$ref === 'string') {
        type = getTypeFromRef(info.$ref)
    }

    if (Array.isArray(info.anyOf)) {
        // assuming 2nd is null, but let's check to ensure
        if (info.anyOf.length !== 2) {
            throw new Error('case not handled by transpiler. contact maintainers.')
        }
        const [nullableType, nullType] = info.anyOf;
        if (nullType?.type !== 'null') {
            throw new Error('[nullableType.type]: case not handled by transpiler. contact maintainers.')
        }
        if (!nullableType?.$ref) {
            if (nullableType.title) {
                type = t.tsTypeReference(t.identifier(nullableType.title));
            } else {
                throw new Error('[nullableType.title] case not handled by transpiler. contact maintainers.')
            }
        } else {
            type = getTypeFromRef(nullableType?.$ref);
        }
        optional = true;
    }

    if (typeof info.type === 'string') {
        if (info.type === 'array') {
            if (info.items.$ref) {
                type = getArrayTypeFromRef(info.items.$ref);
            } else if (info.items.title) {
                type = t.tsArrayType(
                    t.tsTypeReference(
                        t.identifier(info.items.title)
                    )
                );
            } else if (info.items.type) {
                type = getArrayTypeFromType(info.items.type);
            } else {
                throw new Error('[info.items] case not handled by transpiler. contact maintainers.')
            }
        } else {
            type = getType(info.type);
        }
    }

    if (Array.isArray(info.type)) {
        // assuming 2nd is null, but let's check to ensure
        if (info.type.length !== 2) {
            throw new Error('please report this to maintainers (field type): ' + JSON.stringify(info, null, 2))
        }
        const [nullableType, nullType] = info.type;
        if (nullType !== 'null') {
            throw new Error('please report this to maintainers (field type): ' + JSON.stringify(info, null, 2))
        }

        if (nullableType === 'array') {
            type = t.tsArrayType(
                getType(info.items.type)
            );
        } else {
            type = getType(nullableType);
        }

        optional = true;
    }
    if (!type) {
        throw new Error('cannot find type for ' + JSON.stringify(info))
    }

    if (schema.required?.includes(prop)) {
        optional = false;
    }

    return { type, optional };
};

export const createTypedObjectParams = (jsonschema: any, camelize: boolean = true) => {
    const keys = Object.keys(jsonschema.properties ?? {});
    if (!keys.length) return;

    const typedParams = keys.map(prop => {
        if (jsonschema.properties[prop].type === 'object') {
            if (jsonschema.properties[prop].title) {
                return propertySignature(
                    camelize ? camel(prop) : prop,
                    t.tsTypeAnnotation(
                        t.tsTypeReference(t.identifier(jsonschema.properties[prop].title))
                    )
                );
            } else {
                throw new Error('createTypedObjectParams() contact maintainer')
            }
        }

        if (Array.isArray(jsonschema.properties[prop].allOf)) {
            const isOptional = !jsonschema.required?.includes(prop);
            const unionTypes = jsonschema.properties[prop].allOf.map(el => {
                if (el.title) return el.title;
                return el.type;
            });
            const uniqUnionTypes = [...new Set(unionTypes)];

            if (uniqUnionTypes.length === 1) {
                return propertySignature(
                    camelize ? camel(prop) : prop,
                    t.tsTypeAnnotation(
                        t.tsTypeReference(
                            t.identifier(uniqUnionTypes[0])
                        )
                    ),
                    isOptional
                );
            } else {
                return propertySignature(
                    camelize ? camel(prop) : prop,
                    t.tsTypeAnnotation(
                        t.tsUnionType(
                            uniqUnionTypes.map(typ =>
                                t.tsTypeReference(
                                    t.identifier(typ)
                                )
                            )
                        )
                    ),
                    isOptional
                );
            }
        } else if (Array.isArray(jsonschema.properties[prop].oneOf)) {
            const isOptional = !jsonschema.required?.includes(prop);
            const unionTypes = jsonschema.properties[prop].oneOf.map(el => {
                if (el.title) return el.title;
                return el.type;
            });
            const uniqUnionTypes = [...new Set(unionTypes)];
            if (uniqUnionTypes.length === 1) {
                return propertySignature(
                    camelize ? camel(prop) : prop,
                    t.tsTypeAnnotation(
                        t.tsTypeReference(
                            t.identifier(uniqUnionTypes[0])
                        )
                    ),
                    isOptional
                );
            } else {
                return propertySignature(
                    camelize ? camel(prop) : prop,
                    t.tsTypeAnnotation(
                        t.tsUnionType(
                            uniqUnionTypes.map(typ =>
                                t.tsTypeReference(
                                    t.identifier(typ)
                                )
                            )
                        )
                    ),
                    isOptional
                );
            }

        }

        try {
            getPropertyType(jsonschema, prop);
        } catch (e) {
            console.log(e);
            console.log(jsonschema, prop);
        }

        const { type, optional } = getPropertyType(jsonschema, prop);
        return propertySignature(
            camelize ? camel(prop) : prop,
            t.tsTypeAnnotation(
                type
            ),
            optional
        );
    });
    const params = keys.map(prop => {
        return t.objectProperty(
            camelize ? t.identifier(camel(prop)) : t.identifier(prop),
            camelize ? t.identifier(camel(prop)) : t.identifier(prop),
            false,
            true
        );
    });

    const obj = t.objectPattern(
        [
            ...params
        ]
    );
    obj.typeAnnotation = t.tsTypeAnnotation(
        t.tsTypeLiteral(
            [
                ...typedParams
            ]
        )
    );

    return obj;
};
