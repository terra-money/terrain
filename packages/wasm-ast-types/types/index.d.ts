export * from './utils';
export * from './wasm';
export * from './recoil';
export * from './messages';
export * from './react-query';
