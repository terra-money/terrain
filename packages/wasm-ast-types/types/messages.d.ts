import * as t from '@babel/types';
import { ExecuteMsg } from './types';
export declare const createFromPartialClass: (className: string, implementsClassName: string, execMsg: ExecuteMsg) => t.ExportNamedDeclaration;
export declare const createFromPartialInterface: (className: string, execMsg: ExecuteMsg) => t.ExportNamedDeclaration;
