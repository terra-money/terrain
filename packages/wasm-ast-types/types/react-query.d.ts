import * as t from '@babel/types';
import { QueryMsg } from './types';
interface ReactQueryHookQuery {
    hookName: string;
    hookParamsTypeName: string;
    hookKeyName: string;
    responseType: string;
    methodName: string;
    jsonschema: any;
}
export declare const createReactQueryHooks: (queryMsg: QueryMsg, contractName: string, QueryClient: string) => any;
export declare const createReactQueryHook: ({ hookName, hookParamsTypeName, responseType, hookKeyName, methodName, jsonschema }: ReactQueryHookQuery) => t.ExportNamedDeclaration;
interface ReactQueryHookQueryInterface {
    QueryClient: string;
    hookParamsTypeName: string;
    responseType: string;
    jsonschema: any;
}
export declare const createReactQueryHookInterface: ({ QueryClient, hookParamsTypeName, responseType, jsonschema }: ReactQueryHookQueryInterface) => t.ExportNamedDeclaration;
export {};
