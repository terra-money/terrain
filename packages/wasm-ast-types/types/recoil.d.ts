import * as t from '@babel/types';
import { QueryMsg } from './types';
export declare const createWasmRecoilMethod: (jsonschema: any) => void;
export declare const createRecoilSelector: (keyPrefix: string, QueryClient: string, methodName: string) => t.ExportNamedDeclaration;
export declare const createRecoilSelectors: (keyPrefix: string, QueryClient: string, queryMsg: QueryMsg) => any;
export declare const createRecoilQueryClientType: () => {
    type: string;
    id: {
        type: string;
        name: string;
    };
    typeAnnotation: {
        type: string;
        members: {
            type: string;
            key: {
                type: string;
                name: string;
            };
            computed: boolean;
            typeAnnotation: {
                type: string;
                typeAnnotation: {
                    type: string;
                };
            };
        }[];
    };
};
export declare const createRecoilQueryClient: (keyPrefix: string, QueryClient: string) => t.ExportNamedDeclaration;
