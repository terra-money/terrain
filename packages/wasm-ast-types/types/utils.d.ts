import * as t from '@babel/types';
import { Field, QueryMsg, ExecuteMsg } from './types';
import { TSTypeAnnotation, TSExpressionWithTypeArguments } from '@babel/types';
export declare const getMessageProperties: (msg: QueryMsg | ExecuteMsg) => any;
export declare const tsPropertySignature: (key: t.Expression, typeAnnotation: t.TSTypeAnnotation, optional: boolean) => t.TSPropertySignature;
export declare const tsObjectPattern: (properties: (t.RestElement | t.ObjectProperty)[], typeAnnotation: t.TSTypeAnnotation) => t.ObjectPattern;
export declare const callExpression: (callee: t.Expression | t.V8IntrinsicIdentifier, _arguments: (t.Expression | t.SpreadElement | t.ArgumentPlaceholder)[], typeParameters: t.TSTypeParameterInstantiation) => t.CallExpression;
export declare const bindMethod: (name: string) => t.ExpressionStatement;
export declare const typedIdentifier: (name: string, typeAnnotation: TSTypeAnnotation, optional?: boolean) => t.Identifier;
export declare const promiseTypeAnnotation: (name: any) => t.TSTypeAnnotation;
export declare const classDeclaration: (name: string, body: any[], implementsExressions?: TSExpressionWithTypeArguments[], superClass?: t.Identifier) => t.ClassDeclaration;
export declare const classProperty: (name: string, typeAnnotation?: TSTypeAnnotation, isReadonly?: boolean, isStatic?: boolean) => t.ClassProperty;
export declare const arrowFunctionExpression: (params: (t.Identifier | t.Pattern | t.RestElement)[], body: t.BlockStatement, returnType: t.TSTypeAnnotation, isAsync?: boolean) => t.ArrowFunctionExpression;
export declare const recursiveNamespace: (names: any, moduleBlockBody: any) => any;
export declare const arrayTypeNDimensions: (body: any, n: any) => any;
export declare const FieldTypeAsts: {
    string: () => t.TSStringKeyword;
    array: (type: any) => t.TSArrayType;
    Duration: () => t.TSTypeReference;
    Height: () => t.TSTypeReference;
    Coin: () => t.TSTypeReference;
    Long: () => t.TSTypeReference;
};
export declare const shorthandProperty: (prop: string) => t.ObjectProperty;
export declare const importStmt: (names: string[], path: string) => t.ImportDeclaration;
export declare const importAminoMsg: () => t.ImportDeclaration;
export declare const getFieldDimensionality: (field: Field) => {
    typeName: string;
    dimensions: number;
    isArray: boolean;
};
export declare const memberExpressionOrIdentifier: (names: any) => any;
export declare const memberExpressionOrIdentifierSnake: (names: any) => any;
