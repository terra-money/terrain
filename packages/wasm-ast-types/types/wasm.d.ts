import * as t from '@babel/types';
import { QueryMsg, ExecuteMsg } from './types';
export declare const identifier: (name: string, typeAnnotation: t.TSTypeAnnotation, optional?: boolean) => t.Identifier;
export declare const getPropertyType: (schema: any, prop: any) => {
    type: any;
    optional: any;
};
export declare const createWasmQueryMethod: (jsonschema: any) => t.ClassProperty;
export declare const createQueryClass: (className: string, implementsClassName: string, queryMsg: QueryMsg) => t.ExportNamedDeclaration;
export declare const createWasmExecMethod: (jsonschema: any) => t.ClassProperty;
export declare const createExecuteClass: (className: string, implementsClassName: string, extendsClassName: string, execMsg: ExecuteMsg) => t.ExportNamedDeclaration;
export declare const createExecuteInterface: (className: string, extendsClassName: string | null, execMsg: ExecuteMsg) => t.ExportNamedDeclaration;
export declare const propertySignature: (name: string, typeAnnotation: t.TSTypeAnnotation, optional?: boolean) => {
    type: string;
    key: t.Identifier;
    typeAnnotation: t.TSTypeAnnotation;
    optional: boolean;
};
export declare const createTypedObjectParams: (jsonschema: any, camelize?: boolean) => t.ObjectPattern;
export declare const createPropertyFunctionWithObjectParams: (methodName: string, responseType: string, jsonschema: any) => t.TSPropertySignature;
export declare const createPropertyFunctionWithObjectParamsForExec: (methodName: string, responseType: string, jsonschema: any) => t.TSPropertySignature;
export declare const createQueryInterface: (className: string, queryMsg: QueryMsg) => t.ExportNamedDeclaration;
export declare const createTypeOrInterface: (Type: string, jsonschema: any) => t.ExportNamedDeclaration;
export declare const createTypeInterface: (jsonschema: any) => t.ExportNamedDeclaration;
