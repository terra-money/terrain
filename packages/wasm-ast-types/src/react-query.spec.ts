import { importStmt } from './utils'
import generate from '@babel/generator';
import * as t from '@babel/types';

import query_msg from './../../../__fixtures__/basic/query_msg.json';
import execute_msg from './../../../__fixtures__/basic/execute_msg_for__empty.json';
import approval_response from './../../../__fixtures__/basic/approval_response.json';
import all_nft_info_response from './../../../__fixtures__/basic/all_nft_info_response.json';
import approvals_response from './../../../__fixtures__/basic/approvals_response.json';
import collection_info_response from './../../../__fixtures__/basic/collection_info_response.json';
import contract_info_response from './../../../__fixtures__/basic/contract_info_response.json';
import instantiate_msg from './../../../__fixtures__/basic/instantiate_msg.json';
import nft_info_response from './../../../__fixtures__/basic/nft_info_response.json';
import num_tokens_response from './../../../__fixtures__/basic/num_tokens_response.json';
import operators_response from './../../../__fixtures__/basic/operators_response.json';
import owner_of_response from './../../../__fixtures__/basic/owner_of_response.json';
import tokens_response from './../../../__fixtures__/basic/tokens_response.json';

import {
  createReactQueryHook,
  createReactQueryHooks
} from './react-query'

const expectCode = (ast) => {
  expect(
    generate(ast).code
  ).toMatchSnapshot();
}

const printCode = (ast) => {
  console.log(
    generate(ast).code
  );
}

// it('createReactQueryHook', () => {
//   printCode(createReactQueryHook({
//     methodName: 'allNftInfo',
//     hookName: 'useSg721AllNftInfoQuery',
//     hookParamsTypeName: 'Sg721CollectionInfoQuery',
//     responseType: 'AllNftInfoResponse',
//     hookKeyName: 'Sg721AllNftInfo'
//   }))
// });


it('createReactQueryHooks', () => {
  expectCode(t.program(
    createReactQueryHooks(
      query_msg,
      'Sg721',
      'Sg721QueryClient'
    )))
  expectCode(t.program(
    createReactQueryHooks(
      query_msg,
      'Sg721',
      'Sg721QueryClient'
    )))
});

