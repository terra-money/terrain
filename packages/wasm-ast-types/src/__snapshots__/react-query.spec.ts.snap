// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createReactQueryHooks 1`] = `
"export interface Sg721CollectionInfoQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<CollectionInfoResponse, Error, CollectionInfoResponse, (string | undefined)[]>;
}
export function useSg721CollectionInfoQuery({
  client,
  options
}: Sg721CollectionInfoQuery) {
  return useQuery<CollectionInfoResponse, Error, CollectionInfoResponse, (string | undefined)[]>([\\"sg721CollectionInfo\\", client.contractAddress], () => client.collectionInfo(), options);
}
export interface Sg721MinterQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<MinterResponse, Error, MinterResponse, (string | undefined)[]>;
}
export function useSg721MinterQuery({
  client,
  options
}: Sg721MinterQuery) {
  return useQuery<MinterResponse, Error, MinterResponse, (string | undefined)[]>([\\"sg721Minter\\", client.contractAddress], () => client.minter(), options);
}
export interface Sg721AllTokensQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<AllTokensResponse, Error, AllTokensResponse, (string | undefined)[]>;
  args: {
    limit?: number;
    startAfter?: string;
  };
}
export function useSg721AllTokensQuery({
  client,
  args,
  options
}: Sg721AllTokensQuery) {
  return useQuery<AllTokensResponse, Error, AllTokensResponse, (string | undefined)[]>([\\"sg721AllTokens\\", client.contractAddress], () => client.allTokens({
    limit: args.limit,
    startAfter: args.startAfter
  }), options);
}
export interface Sg721TokensQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<TokensResponse, Error, TokensResponse, (string | undefined)[]>;
  args: {
    limit?: number;
    owner: string;
    startAfter?: string;
  };
}
export function useSg721TokensQuery({
  client,
  args,
  options
}: Sg721TokensQuery) {
  return useQuery<TokensResponse, Error, TokensResponse, (string | undefined)[]>([\\"sg721Tokens\\", client.contractAddress], () => client.tokens({
    limit: args.limit,
    owner: args.owner,
    startAfter: args.startAfter
  }), options);
}
export interface Sg721AllNftInfoQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<AllNftInfoResponse, Error, AllNftInfoResponse, (string | undefined)[]>;
  args: {
    includeExpired?: boolean;
    tokenId: string;
  };
}
export function useSg721AllNftInfoQuery({
  client,
  args,
  options
}: Sg721AllNftInfoQuery) {
  return useQuery<AllNftInfoResponse, Error, AllNftInfoResponse, (string | undefined)[]>([\\"sg721AllNftInfo\\", client.contractAddress], () => client.allNftInfo({
    includeExpired: args.includeExpired,
    tokenId: args.tokenId
  }), options);
}
export interface Sg721NftInfoQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<NftInfoResponse, Error, NftInfoResponse, (string | undefined)[]>;
  args: {
    tokenId: string;
  };
}
export function useSg721NftInfoQuery({
  client,
  args,
  options
}: Sg721NftInfoQuery) {
  return useQuery<NftInfoResponse, Error, NftInfoResponse, (string | undefined)[]>([\\"sg721NftInfo\\", client.contractAddress], () => client.nftInfo({
    tokenId: args.tokenId
  }), options);
}
export interface Sg721ContractInfoQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<ContractInfoResponse, Error, ContractInfoResponse, (string | undefined)[]>;
}
export function useSg721ContractInfoQuery({
  client,
  options
}: Sg721ContractInfoQuery) {
  return useQuery<ContractInfoResponse, Error, ContractInfoResponse, (string | undefined)[]>([\\"sg721ContractInfo\\", client.contractAddress], () => client.contractInfo(), options);
}
export interface Sg721NumTokensQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<NumTokensResponse, Error, NumTokensResponse, (string | undefined)[]>;
}
export function useSg721NumTokensQuery({
  client,
  options
}: Sg721NumTokensQuery) {
  return useQuery<NumTokensResponse, Error, NumTokensResponse, (string | undefined)[]>([\\"sg721NumTokens\\", client.contractAddress], () => client.numTokens(), options);
}
export interface Sg721AllOperatorsQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<AllOperatorsResponse, Error, AllOperatorsResponse, (string | undefined)[]>;
  args: {
    includeExpired?: boolean;
    limit?: number;
    owner: string;
    startAfter?: string;
  };
}
export function useSg721AllOperatorsQuery({
  client,
  args,
  options
}: Sg721AllOperatorsQuery) {
  return useQuery<AllOperatorsResponse, Error, AllOperatorsResponse, (string | undefined)[]>([\\"sg721AllOperators\\", client.contractAddress], () => client.allOperators({
    includeExpired: args.includeExpired,
    limit: args.limit,
    owner: args.owner,
    startAfter: args.startAfter
  }), options);
}
export interface Sg721ApprovalsQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<ApprovalsResponse, Error, ApprovalsResponse, (string | undefined)[]>;
  args: {
    includeExpired?: boolean;
    tokenId: string;
  };
}
export function useSg721ApprovalsQuery({
  client,
  args,
  options
}: Sg721ApprovalsQuery) {
  return useQuery<ApprovalsResponse, Error, ApprovalsResponse, (string | undefined)[]>([\\"sg721Approvals\\", client.contractAddress], () => client.approvals({
    includeExpired: args.includeExpired,
    tokenId: args.tokenId
  }), options);
}
export interface Sg721ApprovalQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<ApprovalResponse, Error, ApprovalResponse, (string | undefined)[]>;
  args: {
    includeExpired?: boolean;
    spender: string;
    tokenId: string;
  };
}
export function useSg721ApprovalQuery({
  client,
  args,
  options
}: Sg721ApprovalQuery) {
  return useQuery<ApprovalResponse, Error, ApprovalResponse, (string | undefined)[]>([\\"sg721Approval\\", client.contractAddress], () => client.approval({
    includeExpired: args.includeExpired,
    spender: args.spender,
    tokenId: args.tokenId
  }), options);
}
export interface Sg721OwnerOfQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<OwnerOfResponse, Error, OwnerOfResponse, (string | undefined)[]>;
  args: {
    includeExpired?: boolean;
    tokenId: string;
  };
}
export function useSg721OwnerOfQuery({
  client,
  args,
  options
}: Sg721OwnerOfQuery) {
  return useQuery<OwnerOfResponse, Error, OwnerOfResponse, (string | undefined)[]>([\\"sg721OwnerOf\\", client.contractAddress], () => client.ownerOf({
    includeExpired: args.includeExpired,
    tokenId: args.tokenId
  }), options);
}"
`;

exports[`createReactQueryHooks 2`] = `
"export interface Sg721CollectionInfoQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<CollectionInfoResponse, Error, CollectionInfoResponse, (string | undefined)[]>;
}
export function useSg721CollectionInfoQuery({
  client,
  options
}: Sg721CollectionInfoQuery) {
  return useQuery<CollectionInfoResponse, Error, CollectionInfoResponse, (string | undefined)[]>([\\"sg721CollectionInfo\\", client.contractAddress], () => client.collectionInfo(), options);
}
export interface Sg721MinterQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<MinterResponse, Error, MinterResponse, (string | undefined)[]>;
}
export function useSg721MinterQuery({
  client,
  options
}: Sg721MinterQuery) {
  return useQuery<MinterResponse, Error, MinterResponse, (string | undefined)[]>([\\"sg721Minter\\", client.contractAddress], () => client.minter(), options);
}
export interface Sg721AllTokensQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<AllTokensResponse, Error, AllTokensResponse, (string | undefined)[]>;
  args: {
    limit?: number;
    startAfter?: string;
  };
}
export function useSg721AllTokensQuery({
  client,
  args,
  options
}: Sg721AllTokensQuery) {
  return useQuery<AllTokensResponse, Error, AllTokensResponse, (string | undefined)[]>([\\"sg721AllTokens\\", client.contractAddress], () => client.allTokens({
    limit: args.limit,
    startAfter: args.startAfter
  }), options);
}
export interface Sg721TokensQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<TokensResponse, Error, TokensResponse, (string | undefined)[]>;
  args: {
    limit?: number;
    owner: string;
    startAfter?: string;
  };
}
export function useSg721TokensQuery({
  client,
  args,
  options
}: Sg721TokensQuery) {
  return useQuery<TokensResponse, Error, TokensResponse, (string | undefined)[]>([\\"sg721Tokens\\", client.contractAddress], () => client.tokens({
    limit: args.limit,
    owner: args.owner,
    startAfter: args.startAfter
  }), options);
}
export interface Sg721AllNftInfoQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<AllNftInfoResponse, Error, AllNftInfoResponse, (string | undefined)[]>;
  args: {
    includeExpired?: boolean;
    tokenId: string;
  };
}
export function useSg721AllNftInfoQuery({
  client,
  args,
  options
}: Sg721AllNftInfoQuery) {
  return useQuery<AllNftInfoResponse, Error, AllNftInfoResponse, (string | undefined)[]>([\\"sg721AllNftInfo\\", client.contractAddress], () => client.allNftInfo({
    includeExpired: args.includeExpired,
    tokenId: args.tokenId
  }), options);
}
export interface Sg721NftInfoQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<NftInfoResponse, Error, NftInfoResponse, (string | undefined)[]>;
  args: {
    tokenId: string;
  };
}
export function useSg721NftInfoQuery({
  client,
  args,
  options
}: Sg721NftInfoQuery) {
  return useQuery<NftInfoResponse, Error, NftInfoResponse, (string | undefined)[]>([\\"sg721NftInfo\\", client.contractAddress], () => client.nftInfo({
    tokenId: args.tokenId
  }), options);
}
export interface Sg721ContractInfoQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<ContractInfoResponse, Error, ContractInfoResponse, (string | undefined)[]>;
}
export function useSg721ContractInfoQuery({
  client,
  options
}: Sg721ContractInfoQuery) {
  return useQuery<ContractInfoResponse, Error, ContractInfoResponse, (string | undefined)[]>([\\"sg721ContractInfo\\", client.contractAddress], () => client.contractInfo(), options);
}
export interface Sg721NumTokensQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<NumTokensResponse, Error, NumTokensResponse, (string | undefined)[]>;
}
export function useSg721NumTokensQuery({
  client,
  options
}: Sg721NumTokensQuery) {
  return useQuery<NumTokensResponse, Error, NumTokensResponse, (string | undefined)[]>([\\"sg721NumTokens\\", client.contractAddress], () => client.numTokens(), options);
}
export interface Sg721AllOperatorsQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<AllOperatorsResponse, Error, AllOperatorsResponse, (string | undefined)[]>;
  args: {
    includeExpired?: boolean;
    limit?: number;
    owner: string;
    startAfter?: string;
  };
}
export function useSg721AllOperatorsQuery({
  client,
  args,
  options
}: Sg721AllOperatorsQuery) {
  return useQuery<AllOperatorsResponse, Error, AllOperatorsResponse, (string | undefined)[]>([\\"sg721AllOperators\\", client.contractAddress], () => client.allOperators({
    includeExpired: args.includeExpired,
    limit: args.limit,
    owner: args.owner,
    startAfter: args.startAfter
  }), options);
}
export interface Sg721ApprovalsQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<ApprovalsResponse, Error, ApprovalsResponse, (string | undefined)[]>;
  args: {
    includeExpired?: boolean;
    tokenId: string;
  };
}
export function useSg721ApprovalsQuery({
  client,
  args,
  options
}: Sg721ApprovalsQuery) {
  return useQuery<ApprovalsResponse, Error, ApprovalsResponse, (string | undefined)[]>([\\"sg721Approvals\\", client.contractAddress], () => client.approvals({
    includeExpired: args.includeExpired,
    tokenId: args.tokenId
  }), options);
}
export interface Sg721ApprovalQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<ApprovalResponse, Error, ApprovalResponse, (string | undefined)[]>;
  args: {
    includeExpired?: boolean;
    spender: string;
    tokenId: string;
  };
}
export function useSg721ApprovalQuery({
  client,
  args,
  options
}: Sg721ApprovalQuery) {
  return useQuery<ApprovalResponse, Error, ApprovalResponse, (string | undefined)[]>([\\"sg721Approval\\", client.contractAddress], () => client.approval({
    includeExpired: args.includeExpired,
    spender: args.spender,
    tokenId: args.tokenId
  }), options);
}
export interface Sg721OwnerOfQuery {
  client: Sg721QueryClient;
  options?: UseQueryOptions<OwnerOfResponse, Error, OwnerOfResponse, (string | undefined)[]>;
  args: {
    includeExpired?: boolean;
    tokenId: string;
  };
}
export function useSg721OwnerOfQuery({
  client,
  args,
  options
}: Sg721OwnerOfQuery) {
  return useQuery<OwnerOfResponse, Error, OwnerOfResponse, (string | undefined)[]>([\\"sg721OwnerOf\\", client.contractAddress], () => client.ownerOf({
    includeExpired: args.includeExpired,
    tokenId: args.tokenId
  }), options);
}"
`;
